File: cpe-level-1/python/zj-d980.py
#!/usr/bin/env python 

import sys

lines = sys.stdin.readlines()

T = int(lines[0])

for Case in range(1, T+1):
    print(f"Case {Case}: ", end="")
    a = list(map(int, lines[Case].split()))
    a.sort()
    if a[0] <= 0:
        print("Invalid")
    elif a[0] == a[2]:
        print("Equilateral")
    elif a[0] == a[1] or a[1] == a[2]:
        print("Isosceles")
    else:
        if a[0] + a[1] > a[2]:
            print("Scalene")
        else:
            print("Invalid")

File: cpe-level-1/cpp/zj-d980.cpp

#include <iostream>
#include <algorithm>
using namespace std;

int main() {
    int T;
    long long a[3];
    cin >> T;
    for (int Case = 1; Case <= T; Case++){
        cin >> a[0] >> a[1] >> a[2];
        sort(a, a+3);
        cout << "Case " << Case << ": ";
        if (a[0] <= 0) {
            cout << "Invalid\n";
        } else if (a[0] == a[2]){
            cout << "Equilateral\n";
        } else if (a[0] == a[1] || a[1] == a[2]){
            cout << "Isosceles\n";
        } else {
            if (a[0] + a[1] > a[2]) cout << "Scalene\n";
            else cout << "Invalid\n";
        }
    }
    return 0;
}
